chromium_src/chrome/browser/profiles/delete_profile_helper.cc

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include <string_view>
+
+#include "brave/components/sync/service/brave_sync_service_impl.h"
+#include "chrome/browser/sync/sync_service_factory.h"
+#include "components/sync/service/sync_user_settings.h"
+
+// Include to prevent redefining HasPrimaryAccount method at the header
+#include "components/signin/public/identity_manager/identity_manager.h"
+
+// Include to prevent redefining IdentityManager method at the header
+#include "chrome/browser/signin/identity_manager_factory.h"
+
+class Profile;
+
+namespace {
+
+bool StopSyncIfActive(Profile* profile, const char* func_name) {
+  // If HasPrimaryAccount is used somewhere out of DisableSyncForProfileDeletion
+  // we don't want to stop Sync.
+  if (std::string(func_name) != "DisableSyncForProfileDeletion") {
```

Comment:
can we do string_view here?

Reply:
yes, thanks, fixed https://github.com/brave/brave-core/pull/24988/commits/2f7c6e866046326326519cc2664d552a9e3aaa2e